src/util/validation.cpp

```diff
@@ -5,15 +5,20 @@
 
 #include <util/validation.h>
 
-#include <consensus/validation.h>
 #include <tinyformat.h>
 
-/** Convert ValidationState to a human-readable message for logging */
 std::string FormatStateMessage(const ValidationState &state)
 {
-    return strprintf("%s%s",
-        state.GetRejectReason(),
-        state.GetDebugMessage().empty() ? "" : ", "+state.GetDebugMessage());
+    if (state.IsValid()) {
+        return "Valid";
+    }
+
+    const std::string debug_message = state.GetDebugMessage();
+    if (!debug_message.empty()) {
+        return strprintf("%s, %s", state.GetRejectReason(), debug_message);
+    }
```

Comment:
In commit "Format CValidationState properly in all cases" (ee4d66a38c5c09161f8a3e63047db6b6908687f4)

I'm a little confused by all the changes in this commit. As far as I can tell the only change that affects behavior is adding `if (state.IsValid()) return "Valid";` at the top of this function, and everything else is a rewrite / style cleanup. Not a big deal, but in the future it might be good to fully motivate changes in the commit description or do style changes in separate commits.